int8 direction # 1: up, -1: down
---
bool success
string message
